src/libgetenv/src/getenv.cpp

```diff
@@ -98,7 +98,11 @@ KeySet *elektraDocu = ksNew(20,
 #include "readme_elektrify-getenv.c"
 	KS_END);
 
-pthread_mutex_t elektraGetEnvMutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+#if defined(__APPLE__) && defined(__MACH__)
+	pthread_mutex_t elektraGetEnvMutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
```

Comment:
ohh, is PTHREAD_RECURSIVE_MUTEX_INITIALIZE  really available? I think we can make resolver code much shorter then.